components/kyma-environment-broker/internal/storage/driver/postsql/events/events.go

```diff
@@ -0,0 +1,80 @@
+package events
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage/dbmodel"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage/postsql"
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	Enabled       bool          `envconfig:"default=false"`
+	Retention     time.Duration `envconfig:"default=72h"`
+	PollingPeriod time.Duration `envconfig:"default=1h"`
+}
+
+var (
+	ev       *events
+	initLock sync.Mutex
+)
+
+type events struct {
+	postsql.Factory
+
+	log logrus.FieldLogger
+}
+
+func New(cfg Config, sess postsql.Factory, log logrus.FieldLogger) *events {
+	if !cfg.Enabled {
+		return nil
+	}
+	initLock.Lock()
+	defer initLock.Unlock()
+	if ev == nil {
+		ev = &events{
+			Factory: sess,
+			log:     log,
+		}
+		go ev.gc(cfg)
+	}
+	return ev
+}
+
+func Infof(instanceID, operationID, format string, args ...string) {
+	ev.InsertEvent(dbmodel.InfoEventLevel, fmt.Sprintf(format, args), instanceID, operationID)
+}
+
+func Errorf(instanceID, operationID string, err error, format string, args ...string) {
+	ev.InsertEvent(dbmodel.ErrorEventLevel, fmt.Sprintf("%v: %v", fmt.Sprintf(format, args), err), instanceID, operationID)
+}
+
+func (e *events) InsertEvent(eventLevel dbmodel.EventLevel, message, instanceID, operationID string) {
+	if e != nil {
+		sess := e.NewWriteSession()
+		if err := sess.InsertEvent(eventLevel, message, instanceID, operationID); err != nil {
+			e.log.Errorf("failed to insert event [%v] %v/%v %q: %v", eventLevel, instanceID, operationID, message, err)
+		}
+	} else {
+		log.Printf("no event sink set for event [%v] %v/%v %q\n", eventLevel, instanceID, operationID, message)
```

Comment:
I'm worried about spam in production logs if eventing is disabled

Reply:
good point, probably worth dropping this entirely